### ISSUE TITLE
Support pickling

### ISSUE THREAD

**Issue by gooppe** [VKCOM/YouTokenToMe#32]
Pickling is often necessary using multiprocessing. Can you add such a feature?

```
>>> import pickle
>>> from youtokentome import BPE
>>> bpe = BPE("shared.model")
>>> file = open("test.pkl", "w")
>>> pickle.dump(bpe, file)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "stringsource", line 2, in _youtokentome_cython.BPE.__reduce_cython__
TypeError: self.encoder cannot be converted to a Python object for pickling
>>> 
```

**Comment by gooppe** [comment]
Actually, I solved the problem with multiprocessing. The need for pickling was produced by the pytorch bug when using multiple workers in the data loader.

**Comment by brian8128** [comment]
This would be nice to have even if we can do multiprocessing without it.

**Comment by 1475963** [comment]
Hi, you do this to pickle the BPE object or any object that contains a BPE encoder.

```python
import youtokentome as yttm

def bpe_reduce(self):
	return (
		self.__class__,
		(str(MODEL_PATH),),
	)
yttm.BPE.__reduce__ = bpe_reduce
```

Then  `pickle.dump` any instance of `youtokentome.BPE`.
This will prevent pickle from saving the BPE object, but it will be able to re-instantiate it given the tuple (class, model_path) when you load your pickle file.

**Comment by kefirski** [comment]
Closed in #81
